reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -178,7 +177,14 @@ static Context of(Map<?, ?> map) {
 					entries[4].getKey(), entries[4].getValue());
 			}
 		}
-		return new ContextN(Collections.emptyMap(), map);
+		// Since ContextN(Map) is a low level API that DOES NOT perform null checks,
+		// we need to check every key/value before passing it to ContextN(Map)
+		map.forEach((key, value) -> {
+			Objects.requireNonNull(key, "key");
```

Comment:
let's make the error message a little bit more helpful for once? how abot `"null key found"` and `"null value for key " + key` below?

Reply:
(good catch otherwise)

Reply:
good idea, added